pkgs/stdenv/cross/default.nix

```diff
@@ -0,0 +1,24 @@
+{ system, allPackages, platform, crossSystem, config, ... } @ args:
+
+rec {
+  argClobber = {
+    crossSystem = null;
+    # Ignore custom stdenvs when cross compiling for compatability
+    config = builtins.removeAttrs config [ "replaceStdenv" ];
+  };
+  vanillaStdenv = (import ../. (args // argClobber // {
+    allPackages = args: allPackages (argClobber // args);
+  })).stdenv;
+
+  # Yeah this isn't so cleanly just build-time packages yet. Notice the the
+  # buildPackages <-> stdenvCross cycle. Yup, it's very weird.
+  buildPackages = allPackages {
+    # It's OK to change the built-time dependencies
+    allowCustomOverrides = true;
+    bootStdenv = stdenvCross;
+    inherit system platform crossSystem config;
+  };
+
+  stdenvCross = with buildPackages;
+    makeStdenvCross vanillaStdenv crossSystem binutilsCross gccCrossStageFinal;
```

Comment:
Do not use with, to prevent issues caused by adding names such as `vanillaStdenv` & `crossSystem` at the top-level of nixpkgs set, or from the config override.